Updating node documentation (#762)
citizenfx/natives: PATHFIND/GetVehicleNodeProperties.md

BOOL GET_VEHICLE_NODE_PROPERTIES(float x, float y, float z, int* density, int* flags);
```

## Description
Gets the density and flags of the closest node to the specified position.  

> Density is a value between 0 and 15, indicating how busy the road is.  

Reviewer: ````suggestion
Density is a value between 0 and 15, indicating how busy the road is.  

```c
enum eVehicleNodeProperties {
	OFF_ROAD = 1 << 0,
	ON_PLAYERS_ROAD =  1 << 1,
	NO_BIG_VEHICLES = 1 << 2,
	SWITCHED_OFF = 1 << 3,
	TUNNEL_OR_INTERIOR = 1 << 4,
	LEADS_TO_DEAD_END = 1 << 5,
	HIGHWAY = 1 << 6,
	JUNCTION = 1 << 7,
	TRAFFIC_LIGHT = 1 << 8,
	GIVE_WAY = 1 << 9,
	WATER = 1 << 10,
}
```
````

## Parameters
* **x**: x position of search
* **y**: y position of search
* **z**: z position of search

> * **density**: returned density of traffic ranging from 0 - 15

Reviewer: ```suggestion
* **density**: The traffic density the current node will spawn in a range of 0-15.
```

> * **flags**: returned flags of this node (see below)

Reviewer: ```suggestion
* **flags**: The vehicle node flags, see `eVehicleNodeProperties`.
```

### Flags
* **0x1**:    Offroad

> * **0x2**:    Unknown. Potentially a target for wander tasks as it follows game camera.

Reviewer 2: This has been dynamically tagged as ahead, potentially situated on or close to the road the player is currently on.

* **0x4**:    NoBigVehicles
* **0x8**:    Disabled
* **0x10**:   Tunnel
* **0x20**:   DeadEnd
* **0x40**:   Highway
* **0x80**:   Junction
* **0x100**:  TrafficLight

> * **0x200**:  StopSign

Reviewer 2: Not sure about that one.

> * **0x400**:  Unknown

Reviewer 2: This is the node for the water/boat.

> ### Flags
> * **0x1**:    Offroad
> * **0x2**:    Unknown. Potentially a target for wander tasks as it follows game camera.
> * **0x4**:    NoBigVehicles
> * **0x8**:    Disabled
> * **0x10**:   Tunnel
> * **0x20**:   DeadEnd
> * **0x40**:   Highway
> * **0x80**:   Junction
> * **0x100**:  TrafficLight
> * **0x200**:  StopSign
> * **0x400**:  Unknown

Reviewer: ```suggestion
```

## Return value

> bool indicating whether a node was found

Reviewer 2: Better description would be `Returns false if the node wasn't located or wasn't streamed in, and true otherwise.`

> bool indicating whether a node was found

Reviewer: ```suggestion
Returns `true` if the node was found, or `false` if the node was not found, or was not streamed in.
```

```
Gets the density and flags of the closest node to the specified position.  
Density is a value between 0 and 15, indicating how busy the road is.  

Flags is a bit field:
public enum PathNodeFlags
{
    Offroad = 0x1,
    WanderTarget = 0x2,
    NoBigVehicles = 0x4,
    Disabled = 0x8,

>     Tunnel = 0x10,

Reviewer: Same thing as above, this shouldn't be in a code block

Author: > I don't have much of `CPathFind` documented, so these are my _rough_ notes on this batch of natives. Unsure how valuable they will be to continue your research:
> 
> #### 0x132F52BBA570FE92:
> `p3` looks like a node radius and is used to supplement `n`, likely leading to your "A bit broken as it returns a random edge after `n`" comment. For example, and using your naming convention, here is an outline of one of the distance checks:
> 
> ```c
> if ((nodeForwardLanes + nodeBackwardLanes < n) || distance(node, linkedNode) <= p3)
>     closestDistance = rootManhattanDistance
> else {
>     closestDistance = nodeManhattanDistance // (dy + dx) + (dz * 3.0)
>     ...
> }
> ```
> 
> #### 0x80CA6A8B6C094CC4:
> `nodeType` looks like a bitfield/flag not an explicit type.
> 
> #### 0x0568566ACBB5DEDC:
> Some additional things to check:
> 
> ```
> 0x10  -> Also used in interiors; 'Tunnels' may be a specific instance of this flag.
> 0x20  -> Maybe a combination of "Block If No Lanes" and "Narrowroad"? Grep script usage: iVar4 & 32 == 0 && iVar4 & 4 == 0?
> 0x100 -> Stop/TrafficLightStop
> 0x200 -> Intersection/Crosswalk
> 0x400 -> Water
> ```
> 
> If you are using Codewalker, here are the `YndNode` mappings (I think):
> 
> ```cs
> 0x1 = ((RawData.Flags0 & 8) != 0)
> 0x2 = ((RawData.Flags0 & 0x10) != 0)
> 0x4 = ((RawData.Flags0 & 0x20) != 0)
> 0x8 = ((RawData.Flags2 & 0x80) != 0)
> 0x10 = ((RawData.Flags3 & 1) != 0)
> 0x20 = ((RawData.Flags4 & 0x70) != 0)
> 0x40 = ((RawData.Flags2 & 0x40) != 0 && (RawData.Flags2 & 0x20) == 0)
> 0x80 = ((RawData.LinkCountFlags & 0x1) != 0)
> 0x100 = ((RawData.Flags1 & 0xF8) == 0x78)
> 0x200 = ((RawData.Flags1 & 0xF8) == 0x80)
> 0x400 = ((RawData.Flags2 & 0x20) != 0)
> ```

Thanks for replying. I've actually done a fair bit of extensive research on them that was never merged (need to touch base with dexy to figure out where it is). It was all figured out by cross referencing the incomplete `paths.xml` so the flags names should be accurate. `Junction` is also `Flags2 & 4`. I've yet to push that up.

https://github.com/dexyfex/CodeWalker/pull/70/files
